cashu/wallet/wallet.py

```diff
@@ -486,20 +488,41 @@ async def split(
 
     async def pay_lightning(self, proofs: List[Proof], invoice: str):
         """Pays a lightning invoice"""
-        status = await super().pay_lightning(proofs, invoice)
-        if status["paid"] == True:
+
+        # generate outputs for the change for overpaid fees
+        # we will generate four blanked outputs that the mint will
+        # imprint with value depending on the fees we overpaid
+        n_return_outputs = 4
+        secrets = [self._generate_secret() for _ in range(n_return_outputs)]
+        outputs, rs = self._construct_outputs(n_return_outputs * [1], secrets)
+
+        status = await super().pay_lightning(proofs, invoice, outputs)

```

Comment:
Wallet: Generate blank outputs here (4 outputs). 

The amount (`n_return_outputs * [1]`) is arbitrary (can be anything) and will be overwritten by the mint.